### ISSUE TITLE
Any results?

### ISSUE THREAD

**Issue by NoamGit** [HPI-DeepLearning/DIRNet#3]
Hi,

Thanks for the great code. Just made the "train" work (not evaluated yet), however, I've got some questions:
- Did you run any benchmarks? Any good results?
- Why both `mxnet` and `tensorflow`? Did the mxnet implementation yielded good results?
- in the **mxnet** code:
     - Does the code work for both affine and non-rigid?
     - It seems that you can only train with this specified shape - `(1, 1, img_shape[2], img_shape[3])`.Can I train in mini-batches too? (I think it is in the first dim)
     - Why did you use customsgd and not Adam (as in the paper)?
     - Why changed the `ncc` to  `rmse`?

**Comment by adrianloy** [comment]
Hi,

Not sure if things changed, but back then Mxnet only supported affine transformations. Maybe there are different ones supported now. This is the reason why we didnt really see good results with the Mxnet implementation, as an affine transformation didnt really help much for our data. And its why there is also a tensorflow implementation.

Yeah, looks like we used a mini-batch size of 1. Cant remember if there was a specific reason for it. Probably it just made things easier. The code shouldnt be difficult to adjust to different batch sizes though.

Not sure why we used customsgd. I guess because there are/were no other optimizers implemented for the simple bind API and I didnt want to implement Adam and the customSGD code was copy&pasted from somewhere.

Ncc often produced NaN values if I remember correctly, which caused problems for the training process. Because there was no variance, maybe because of the mini batch size of 1.

Note that this was a short-term student project more than a year ago. So we couldn't try too many different things and I guess code quality isnt the best aswell. Nevertheless, the network worked e.g. for Mnist data. But affine transformation is quite limited and might not be helpful for your data. You should check if the SpatialTransformer layer of mxnet is now able to perform non-rigid transformations.
